generator/src/generator/namespace.rs

```diff
@@ -3918,24 +3919,52 @@ impl<'ns, 'c> NamespaceGenerator<'ns, 'c> {
                 outln!(out, "std::mem::swap(fds, &mut {});", rust_field_name);
             }
             xcbdefs::FieldDef::Expr(expr_ref) => {
-                // The only expr field we ever need to parse is odd_length,
-                // so assert that we have that and then just write some one-off
-                // code for it.
-                assert_eq!(expr_ref.name, "odd_length");
-                outln!(
-                    out,
-                    "let (odd_length, remaining) = bool::try_parse(remaining)?;"
-                );
-                // And there is a matching special case to use this in request parsing.
+                match expr_ref.expr {
+                    xcbdefs::Expression::Value(_) => {
+                        // Treat this as a normal field, that we'll validate in
+                        // emit_field_post_parse.
+                        let rust_field_name = to_rust_variable_name(&expr_ref.name);
+                        outln!(
+                            out,
+                            "let ({}, remaining) = {};",
+                            rust_field_name,
+                            self.emit_value_parse(&expr_ref.type_, from),
+                        );
+                    }
+                    _ => {
+                        // The only non-constant expr field we ever need to parse is
+                        // odd_length, so assert that we have that and then just write
+                        // some one-off code for it.
+                        assert_eq!(expr_ref.name, "odd_length");
+                        outln!(
+                            out,
+                            "let (odd_length, remaining) = bool::try_parse(remaining)?;"
+                        );
+                        // And there is a matching special case to use this in request parsing.
+                    }
+                }
             }
             xcbdefs::FieldDef::VirtualLen(_) => {}
         }
     }
 
     fn emit_field_post_parse(&self, field: &xcbdefs::FieldDef, out: &mut Output) {
-        if let xcbdefs::FieldDef::Normal(normal_field) = field {
-            let rust_field_name = to_rust_variable_name(&normal_field.name);
-            self.emit_value_post_parse(&normal_field.type_, &rust_field_name, out);
+        match field {
+            xcbdefs::FieldDef::Normal(normal_field) => {
+                let rust_field_name = to_rust_variable_name(&normal_field.name);
+                self.emit_value_post_parse(&normal_field.type_, &rust_field_name, out);
+            }
+            xcbdefs::FieldDef::Expr(xcbdefs::ExprField {
+                name,
+                expr: xcbdefs::Expression::Value(v),
+                ..
+            }) => {
+                let rust_field_name = to_rust_variable_name(name);
+                outln!(out, "if {} as u32 != {} {{", rust_field_name, v);
```

Comment:
Why the cast to `u32`? It seems unnecessary to me.

Reply:
It's unnecessary for 0 or 1 but Expression::Value contains a u32.

Reply:
I suppose we could copy the u8::try_from assertion from elsewhere in the patch and guard, and then eliminate the cast.

Reply:
This is "just a number in the source code". The compiler would warn if the number is out of range: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=52f1164ed82fc8c3fd51eda039d713d9